src/app/shared/ubs-add-address-pop-up/ubs-add-address-pop-up.component.ts

```diff
@@ -446,27 +450,18 @@ export class UBSAddAddressPopUpComponent implements OnInit, OnDestroy, AfterView
   selectCity(event: Event): void {
     const cityIndex = (event.target as HTMLSelectElement)?.selectedIndex;
     this.onCitySelected(this.KyivRegion[cityIndex ?? 0]);
-    if (this.currentLanguage === 'ua') {
-      const elem = this.cities.find((el) => {
-        if (el.key <= 10) {
-          return el.cityName === (event.target as HTMLSelectElement).value.slice(3) ? el : undefined;
-        } else {
-          return el.cityName === (event.target as HTMLSelectElement).value.slice(4) ? el : undefined;
-        }
-      });
-      const city = this.cities.find((el) => el.key === elem.key);
-      this.cityEn.setValue(city.cityName);
-    } else {
-      const elem = this.listOflocations.getCity('en').find((el) => {
-        if (el.key <= 10) {
-          return el.cityName === (event.target as HTMLSelectElement).value.slice(3) ? el : undefined;
-        } else {
-          return el.cityName === (event.target as HTMLSelectElement).value.slice(4) ? el : undefined;
-        }
-      });
-      const city = this.cities.find((el) => el.key === elem.key);
-      this.city.setValue(city.cityName);
-    }
+    const elem = this.cities.find((el) => {
+      if (el.key <= 10) {
+        return el.cityName === (event.target as HTMLSelectElement).value.slice(3) ? el : undefined;
```

Comment:
I would suggest a slight improvement just to avoid code duplication:
value.slice(el.key <= 10 ? 3 : 4)